sqlmesh/core/schema_loader.py

```diff
@@ -8,50 +8,66 @@
 from sqlglot import exp
 from sqlglot.dialects.dialect import DialectType
 
+from sqlmesh.core.dialect import normalize_model_name
 from sqlmesh.core.engine_adapter import EngineAdapter
-from sqlmesh.core.model import Model
+from sqlmesh.core.model.definition import Model
 from sqlmesh.core.state_sync import StateReader
-from sqlmesh.utils import yaml
+from sqlmesh.utils import UniqueKeyDict, yaml
 
 logger = logging.getLogger(__name__)
 
 
 def create_schema_file(
     path: Path,
-    models: t.Dict[str, Model],
+    models: UniqueKeyDict[str, Model],
     adapter: EngineAdapter,
     state_reader: StateReader,
     dialect: DialectType,
+    default_catalog: t.Optional[str],
     max_workers: int = 1,
 ) -> None:
     """Create or replace a YAML file with model schemas.
 
     Args:
         path: The path to store the YAML file.
-        models: A dictionary of models to fetch columns from the db.
+        models: FQN to model
         adapter: The engine adapter.
         state_reader: The state reader.
         dialect: The dialect to serialize the schema as.
         max_workers: The max concurrent workers to fetch columns.
     """
-    external_tables = set()
+    external_model_fqns = set()
+    external_model_names = set()
```

Comment:
why do you have to keep track of both?

Reply:
We lookup based on models names but we ultimately need model fqns.

Reply:
this whole section is a bit problematic, i think we need to rethink it. you're also double normalizing it on line 54, since depends_on already does normalization.

lets get rid of the extra set external_model_names, we can check what the original model name is based on the new property you're storing. the extra set is not necessary, we can just build this list after this loop.